### PR TITLE
Added `-h` Flag To All The SDK Commands

### PR DESCRIPTION
**Pull request by shmuel44**
## Related Issues
fixes: [link](https://jira-dc.paloaltonetworks.com/browse/CIAC-10935)

## Description
Added `-h` option as a short option for the `--help` flag to all the Demisto-SDK commands **that use `typer` lib** (and not click)

By adding 
`app = typer.Typer(context_settings={"help_option_names": ["-h", "--help"]})`

### CODE REVIEW

File: .changelog/4350.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Added `-h` option as a short option for the `--help` flag to all the Demisto-SDK commands.
```
Comment (shmuel44) [inline .changelog/4350.yml:2]:
```suggestion
- description: Added the `-h` option as a short option for the `--help` flag to all the type-based commands.

File: .changelog/4350.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Added the `-h` option as a short option for the `--help` flag to all the type-based commands.
```
Comment (shmuel44) [inline .changelog/4350.yml:2]:
```suggestion
- description: Added the `-h` option as a short option for the `--help` flag to all the typer-based commands.
```

File: .changelog/4350.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Added the `-h` option as a short option for the `--help` flag to all the typer-based commands.
```
Comment (dorschw) [inline .changelog/4350.yml:2]:
```suggestion
- description: Added the `-h` option as a short option for the `--help` flag in `typer`-based commands.
```

File: demisto_sdk/__main__.py [hunk 4]
```diff
@@ -3783,7 +3785,10 @@ def pre_commit(
 
 # ====================== modeling-rules command group ====================== #
 modeling_rules_app = typer.Typer(
-    name="modeling-rules", hidden=True, no_args_is_help=True
+    name="modeling-rules",
+    hidden=True,
+    no_args_is_help=True,
```
Comment (samuelFain) [inline demisto_sdk/__main__.py:3790]: Why are some commands flagged with `no_args_is_help=True` or `hidden=True` while others are not? How do we decide which commands receive these flags?

**Review by samuelFain — APPROVED** [review]
Very nice!
Please attach a relevant Jira ticket if such exist.

File: demisto_sdk/__main__.py (hunk 4 above)
Comment (shmuel44) [reply demisto_sdk/__main__.py:3790]: idk, it's not part of my PR
this is the docs for [`no_args_is_help`](https://typer.tiangolo.com/tutorial/commands/#show-the-help-message-if-no-command-is-given)

**Comment by shmuel44** [comment]
> Very nice! Please attach a relevant Jira ticket if such exist.

I will open a ticket

**Pull request merged**